test/images/agnhost/connect/connect.go

```diff
@@ -125,3 +128,44 @@ func connectSCTP(dest string, timeout time.Duration) {
 		os.Exit(1)
 	}
 }
+
+func connectUDP(dest string, timeout time.Duration) {
+	if _, err := net.ResolveUDPAddr("udp", dest); err != nil {
+		fmt.Fprintf(os.Stderr, "DNS: %v\n", err)
+		os.Exit(1)
+	}
+
+	conn, err := net.Dial("udp", dest)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+		os.Exit(1)
+	}
+
+	if timeout > 0 {
+		if err = conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+			fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
+			os.Exit(1)
+		}
+	}
+
+	if _, err = conn.Write([]byte("hostname\n")); err != nil {
+		parseUDPErrorAndExit(err)
+	}
+
+	var buf = make([]byte, 1024)
```

Comment:
we have to verify that we actually received some data
https://github.com/kubernetes/kubernetes/blob/bae644e85e085ce668c3edf23e8789fb623331b4/test/e2e/network/service.go#L536-L561

Reply:
raising an error when `readBytes == 0`.. 
looks something that can be controlled on a flag i.e. `--wait-response`. or even use a `--udp-request` | `--udp-response`..

Reply:
if you don't receive any bytes you can not known that the other side is receiving packets.
This is the same we discussed in slack regarding netcat, it just checks that the packet goes out, it doesn't mean it has been received, there is no ACK... :shrug:

Reply:
maybe you can return an error more descriptive "NO DATA RECEIVED" , and clarifiy that we can't guarantee the port is open